### PR TITLE
Implemented selectbox/searchbar for users to search for a county

### PR DESCRIPTION
**Pull request by owenchen04**
## Overview

Consolidated separate selectors for state and county into a single selectbox

## Replication

Run requirements.txt file. Run the Streamlit app using 'streamlit run searchbox.py' and start typing desired county and/or state, and wait for search suggestions to filter.

## Changes Made

Got rid of original (separate) selector widgets for state and county, combining the functionality into a single selectbox. The selectbox is automatically populated with suggestions from state_counties_corrected.csv in the format 'COUNTY, STATE_ABBREVIATION', and is updated upon user input to suggest counties matching the search term.

## Test Coverage

We manually tested various county inputs in the selectbox.

## Next Steps 

We found several instances where we got a 'EmptyDataError: No columns to parse from file' error on certain user inputs (particularly for small counties). Although we didn't have time to do this yet, we plan to include additional try-excepts to handle these instances in our code in the future. We also began looking at how to style the widget according to the Figma design, and found that there was no easy method on using CSS to style native Streamlit components. We want to try out / confirm the best way to go about styling before we commit to a specific method.

## Related PRs or Issues

This PR iterates off of our last week's work on Issue 'SBU Data Extraction', with related PR 'Created selector widgets for state and county for querying SBU database'. We created a new file starting from a copy of get_sbu_data.py to work off of.

## Screenshots

<details>

  <summary>Example of fuzzy search in select box</summary>

image

  <summary>Example of working user input</summary>

  image

</details>

### CODE REVIEW

**Review by LiuBrianna — COMMENTED** [review]
Nice work on your ticket! 

I saw in your requirements.txt there are some modules that are not used in searchbox.py such as seaborn. I wonder if this module is needed and if not, it should be removed. But I understand if there are some linked dependencies. 

In terms of style, there are comments that are not necessary or confusing such as line 130.  Additionally, there is also line 51-52. It seems like an unneeded comment since you created the sql string and called the data later on. 

In terms of styling, your docstrings for functions are of different layout. This may confuse future users. Please keep one consistent doctoring layout. One example of this is that one function describes the parameters and the returns while another describes what the function does using examples.

File: searchbox.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    county_words = selected_county.split(' ')
+    for word in county_words:
+        res += word.lower().capitalize() + ' '
+    return res[:-1]
+
+county_name = format_county(selected_county)
+
+
+st.header('Get Facilities in Selected County')
+#sql = 'select * from "ECHO_EXPORTER" where "FAC_COUNTY" = \'TOMPKINS\' and "FAC_STATE" = \'NY\'' # "ECHO_EXPORTER" contains basic info about all regulated industrial facilities, including location
```
Comment (jessicajcho) [inline searchbox.py]: can delete this since it's hardcoded to tompkins

**Comment by owenchen04** [comment]
> Nice work on your ticket!
> 
> I saw in your requirements.txt there are some modules that are not used in searchbox.py such as seaborn. I wonder if this module is needed and if not, it should be removed. But I understand if there are some linked dependencies.
> 
> In terms of style, there are comments that are not necessary or confusing such as line 130. Additionally, there is also line 51-52. It seems like an unneeded comment since you created the sql string and called the data later on.
> 
> In terms of styling, your docstrings for functions are of different layout. This may confuse future users. Please keep one consistent doctoring layout. One example of this is that one function describes the parameters and the returns while another describes what the function does using examples.

File: searchbox.py (hunk 1 above)
Comment (owenchen04) [reply searchbox.py]: will do!

**Comment by owenchen04** [comment]
> I saw in your requirements.txt there are some modules that are not used in searchbox.py such as seaborn. I wonder if this module is needed and if not, it should be removed. But I understand if there are some linked dependencies.
> 

I believe some of these dependencies are linked as Anya and I were having a lot of trouble getting these resolved, so I'm hesitant to touch these for the time being (It's also possible that some of these were required in the H4I_EEW_Industry_Reports repo, as that's what we were coding in before switching to EDGI_Website).
 
> In terms of styling, your docstrings for functions are of different layout. This may confuse future users. Please keep one consistent doctoring layout. One example of this is that one function describes the parameters and the returns while another describes what the function does using examples.

Since we're adapting a lot of this from old code, we did not write the majority of these docstrings (with the exception of format_county). While I agree it's confusing, I don't necessarily want to change these specs too much as we want them to communicate the same info as the original authors intended. That said, I have made some slight adjustments (adding some blank lines, making capitalization consistent) to improve readability + consistency.

**Pull request closed without merging**